www/__tests__/enketoHelper.test.ts

```diff
@@ -0,0 +1,245 @@
+ * rarely, so loading it every time will likely do a bunch of unnecessary work.
+ * Loading it on demand seems like the way to go. If we choose to experiment
+ * with incremental updates, we may want to revisit this.
+ */
+it('loads the previous response to a given survey', () => {
+  expect(loadPreviousResponseForSurvey('manual/demographic_survey')).resolves.toMatchObject({
+    data: 'completed',
+    time: '01/01/2001',
+  });
+});
+
+/**
+ * filterByNameAndVersion filter the survey responses by survey name and their version.
+ * The version for filtering is specified in enketo survey `compatibleWith` config.
+ * The stored survey response version must be greater than or equal to `compatibleWith` to be included.
+ */
+it('filters the survey responses by their name and version', () => {
+  //no response -> no filtered responses
+  expect(filterByNameAndVersion('TimeUseSurvey', [])).resolves.toStrictEqual([]);
+
+  const response = [
+    {
+      data: {
+        label: 'Activity', //display label (this value is use for displaying on the button)
+        ts: '100000000', //the timestamp at which the survey was filled out (in seconds)
+        fmt_time: '12:36', //the formatted timestamp at which the survey was filled out
+        name: 'TimeUseSurvey', //survey name
+        version: '1', //survey version
+        xmlResponse: '<this is my xml>', //survey response XML string
+        jsonDocResponse: 'this is my json object', //survey response JSON object
+      },
+      metadata: {},
+    },
+  ];
+
+  //one response -> that response
+  expect(filterByNameAndVersion('TimeUseSurvey', response)).resolves.toStrictEqual(response);
+
+  const responses = [
+    {
+      data: {
+        label: 'Activity', //display label (this value is use for displaying on the button)
+        ts: '100000000', //the timestamp at which the survey was filled out (in seconds)
+        fmt_time: '12:36', //the formatted timestamp at which the survey was filled out
+        name: 'TimeUseSurvey', //survey name
+        version: '1', //survey version
+        xmlResponse: '<this is my xml>', //survey response XML string
+        jsonDocResponse: 'this is my json object', //survey response JSON object
+      },
+      metadata: {},
+    },
+    {
+      data: {
+        label: 'Activity', //display label (this value is use for displaying on the button)
+        ts: '100000000', //the timestamp at which the survey was filled out (in seconds)
+        fmt_time: '12:36', //the formatted timestamp at which the survey was filled out
+        name: 'OtherSurvey', //survey name
+        version: '1', //survey version
+        xmlResponse: '<this is my xml>', //survey response XML string
+        jsonDocResponse: 'this is my json object', //survey response JSON object
+      },
+      metadata: {},
+    },
+  ];
+
+  //several responses -> only the one that has a name match
+  expect(filterByNameAndVersion('TimeUseSurvey', responses)).resolves.toStrictEqual(response);
```

Comment:
I see that responses are filtered if they aren't for TimeUseSurvey.
The more important thing to test here, though, is version compatibility

Reply:
I just added a case in [100eefb](https://github.com/e-mission/e-mission-phone/pull/1063/commits/100eefb76dd7bd118be1f78b8dd003a11c924ac2)